docs/src/components/home/CoreShowcase.tsx

```diff
@@ -0,0 +1,343 @@
+          return [...final.slice(0, final.length - 1), `${final[final.length - 1]} `];
+        }
+        return [...final, curr];
+      }, [] as Array<string>);
+
+      const time = setInterval(() => {
+        if (index.current <= code.length - 1) {
+          setResult((current) => `${current}${code[index.current]}`);
+        }
+        index.current += 1;
+        if (index.current >= code.length) {
+          onStepComplete(step);
+          clearInterval(time);
+        }
+      }, 50);
+      return () => {
+        clearInterval(time);
+      };
+    }
+    setResult('');
+    return () => {};
+  }, [codes, step, onStepComplete]);
+  return <HighlightedCode component={MarkdownElement} code={result} language="jsx" />;
+}
+
+export default function CoreShowcase() {
+  const totalStep = CODES.length;
+  const theme = useTheme();
+  const codeContainer = React.useRef<HTMLDivElement | null>(null);
+  const [step, setStep] = React.useState(0);
+  const [running, setRunning] = React.useState(false);
+  const [customized, setCustomized] = React.useState(false);
+  const codeImports = materialDemoCode.imports(customized);
+  const cursor = React.useRef(0);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const themeFrames = React.useMemo(() => getMaterialThemeFrames(theme), []);
+  const [customTheme, setCustomTheme] = React.useState(defaultTheme);
+  const updateTheme = React.useMemo(
+    () =>
+      function handleUpdateTheme(latestStep: number) {
+        if (latestStep === totalStep) {
+          setRunning(false);
+        }
+        cursor.current += CODES[latestStep - 1].split('\n').length - 1;
+        const themeOptions = produceThemeOptions(themeFrames, latestStep);
+        setCustomTheme(createTheme(themeOptions));
+        setTimeout(() => {
+          setStep((current) => (current < totalStep ? current + 1 : current));
+        }, 1500);
+      },
+    [themeFrames, totalStep],
+  );
+  React.useEffect(() => {
+    if (customized && codeContainer.current) {
+      codeContainer.current.scrollTop = (cursor.current + codeImports.split('\n').length - 1) * 20;
+    }
+    if (step === 0 && customized) {
+      cursor.current = 0;
+      setRunning(true);
+      setCustomTheme(createTheme());
+      const time = setTimeout(() => {
+        setStep(1);
+      }, 200);
+      return () => {
+        clearTimeout(time);
+      };
+    }
+    return () => {};
+  }, [step, customized, codeImports]);
+  React.useEffect(() => {
+    if (customized) {
+      setStep(0);
+      setRunning(true);
+      setCustomTheme(createTheme());
+      const time = setTimeout(() => {
+        setStep(1);
+      }, 200);
+      return () => {
+        clearTimeout(time);
+      };
+    }
+    setCustomTheme(createTheme());
+    return () => {};
+  }, [customized]);
+  return (
+    <ShowcaseContainer
+      preview={
+        <ThemeProvider theme={customTheme}>
+          <Box sx={{ minWidth: 300, width: '50%', maxWidth: '100%' }}>
```

Comment:
```suggestion
          <Box sx={{ minWidth: 300, width: '80%', maxWidth: '100%' }}>
```
50% gets the component too tight.